src/hex.ts

```diff
@@ -55,6 +65,50 @@ export function assertIsStrictHexString(value: unknown): asserts value is Hex {
   );
 }
 
+/**
+ * Validate that the passed prefixed hex string is a valid hex address, or a
+ * valid mixed-case checksum address.
+ *
+ * @param possibleAddress - Input parameter to check against.
+ * @returns Whether or not the input is a valid hex address.
+ */
+export function isValidHexAddress(possibleAddress: Hex) {
+  return (
+    is(possibleAddress, HexAddressStruct) ||
+    isValidChecksumAddress(possibleAddress)
+  );
+}
+
+/**
+ * Validate that the passed hex string is a valid ERC-55 mixed-case
+ * checksum address.
+ *
+ * @param possibleChecksum - The hex address to check.
+ * @returns True if the address is a checksum address.
+ */
+export function isValidChecksumAddress(possibleChecksum: Hex) {
+  if (!is(possibleChecksum, HexChecksumAddressStruct)) {
+    return false;
+  }
+
+  const unPrefixed = remove0x(possibleChecksum);
+  const unPrefixedHash = remove0x(
+    bytesToHex(keccak256(unPrefixed.toLowerCase())),
+  );
+
+  for (let i = 0; i < unPrefixedHash.length; i++) {
+    const value = parseInt(unPrefixedHash[i] as string, 16);
+    if (
+      (value > 7 && unPrefixed[i]?.toUpperCase() !== unPrefixed[i]) ||
```

Comment:
It would be great to have inline comments to explain what's going on here, and [a link to the EIP](https://eips.ethereum.org/EIPS/eip-55) perhaps. This might look pretty strange to the reader otherwise.

Reply:
do you think that as we now have 
> @see https://eips.ethereum.org/EIPS/eip-55

on the `getChecksumAddress` function it should be easier to guess what is happening, or should we add more information through comments on code?

Reply:
This seems OK now